promise.js

```diff
@@ -0,0 +1,48 @@
+/*jslint browser: true, node: true, maxlen: 128 */
+/*global self */
+
+"use strict";
+
+var globalObject, hasPromiseSupport;
+
+function isFunction(x) {
+    return typeof x === "function";
+}
+
+// Seek the global object
+if (global !== undefined) {
+    globalObject = global;
+} else if (window !== undefined && window.document) {
+    globalObject = window;
+} else {
+    globalObject = self;
+}
```

Comment:
You do not need any of these tests. Just go ahead and use `global` directly. It has existed in node since the beginning. `browserify` sets the value of `global` to `window` anyways, so it will exist even in the browser.